lib/solaar/ui/window.py

```diff
@@ -594,12 +594,16 @@ def _update_receiver_panel(receiver, panel, buttons, full=False):
 	# b._insecure.set_visible(False)
 	buttons._unpair.set_visible(False)
 
-	may_pair = ( receiver.may_unpair or receiver.re_pairs ) and not is_pairing and \
-		( receiver.remaining_pairings() is None or receiver.remaining_pairings() != 0 )
-	if may_pair and not receiver.re_pairs and devices_count >= receiver.max_devices:
-		paired_devices = tuple(n for n in range(1, receiver.max_devices+1) if n in receiver)
-		may_pair &= len(paired_devices) < receiver.max_devices
-	buttons._pair.set_sensitive(may_pair)
+	if ( receiver.may_unpair or receiver.re_pairs ) and not is_pairing and \
+	   ( receiver.remaining_pairings() is None or receiver.remaining_pairings() != 0 ):
+		if not receiver.re_pairs and devices_count >= receiver.max_devices:
+			paired_devices = tuple(n for n in range(1, receiver.max_devices+1) if n in receiver)
+			buttons._pair.set_sensitive(len(paired_devices) < receiver.max_devices)
+		else:
+			buttons._pair.set_sensitive(True)
+	else:
+		buttons._pair.set_sensitive(False)
```

Comment:
Hum, maybe we can move this to a `can_pair` function?

Reply:
The logic is only needed here so I'm not sure that anything would be gained by moving this elsewhere.

Reply:
Sure.